### PR TITLE
[FLINK-24455][tests]FallbackAkkaRpcSystemLoader should check for mave…

### PR DESCRIPTION
**Pull request by Aitozi**
…n errors

## What is the purpose of the change

Add check for in `FallbackAkkaRpcSystemLoader` when download dependencies failed.

### CODE REVIEW

**Comment by Aitozi** [comment]
ping @zentol do you have time to help review this :)

File: flink-rpc/flink-rpc-akka-loader/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcSystemLoader.java [hunk 1]
```diff
@@ -42,6 +43,15 @@
  */
 public class AkkaRpcSystemLoader implements RpcSystemLoader {
 
+    /**
+     * The name of the akka dependency jar, which should be correspond to the artifact output copied
+     * to flink-runtime module.
```
Comment (zentol) [inline flink-rpc/flink-rpc-akka-loader/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcSystemLoader.java:48]:
```suggestion
     * The name of the akka dependency jar, which must match the jar that is bundled by the flink-rpc-akka-loader module.
```
We no longer copy a jar to flink-runtime.

File: flink-rpc/flink-rpc-akka-loader/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcSystemLoader.java [hunk 2]
```diff
@@ -42,6 +43,15 @@
...
+     */
+    private static final String FLINK_RPC_AKKA_FAT_JAR = "flink-rpc-akka.jar";
+
+    protected static final String HINT_USAGE =
```
Comment (zentol) [inline flink-rpc/flink-rpc-akka-loader/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcSystemLoader.java:52]:
```suggestion
    static final String HINT_USAGE =
```
Seems more appropriate.

File: flink-rpc/flink-rpc-akka-loader/src/test/java/org/apache/flink/runtime/rpc/akka/FallbackAkkaRpcSystemLoader.java [hunk 3]
```diff
@@ -67,7 +71,13 @@ public RpcSystem loadRpcSystem(Configuration config) {
                     akkaRpcModuleDirectory.resolve(Paths.get("target", "dependencies"));
 
             if (!Files.exists(akkaRpcModuleDependenciesDirectory)) {
-                downloadDependencies(akkaRpcModuleDirectory, akkaRpcModuleDependenciesDirectory);
+                int exitCode =
+                        downloadDependencies(
+                                akkaRpcModuleDirectory, akkaRpcModuleDependenciesDirectory);
+                if (exitCode != 0) {
+                    throw new RuntimeException(
```
Comment (zentol) [inline flink-rpc/flink-rpc-akka-loader/src/test/java/org/apache/flink/runtime/rpc/akka/FallbackAkkaRpcSystemLoader.java:78]:
```suggestion
                    throw new RpcLoaderException(
```

File: flink-rpc/flink-rpc-akka-loader/src/test/java/org/apache/flink/runtime/rpc/akka/FallbackAkkaRpcSystemLoader.java [hunk 4]
```diff
@@ -67,7 +71,13 @@ public RpcSystem loadRpcSystem(Configuration config) {
...
+                            "Can not download dependencies, please see the output log for reason.");
```
Comment (zentol) [inline flink-rpc/flink-rpc-akka-loader/src/test/java/org/apache/flink/runtime/rpc/akka/FallbackAkkaRpcSystemLoader.java:79]:
```suggestion
                            "Could not download dependencies of flink-rpc-akka, please see the log output for details.");
```

**Review by zentol — CHANGES_REQUESTED** [review]

**Comment by Aitozi** [comment]
Thanks for help review, I have addressed the comments @zentol

**Comment by zentol** [comment]
@flinkbot run azure

**Comment by Aitozi** [comment]
@flinkbot run azure

**Pull request merged**